packages/components/src/ui/radix-dropdown/index.tsx

```diff
@@ -0,0 +1,177 @@
+/**
+ * External dependencies
+ */
+import * as DropdownMenuPrimitive from '@radix-ui/react-dropdown-menu';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef } from '@wordpress/element';
+import { SVG, Circle } from '@wordpress/primitives';
+import { check, lineSolid } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import Icon from '../../icon';
+import * as DropdownMenuStyled from './styles';
+import type {
+	DropdownMenuProps,
+	DropdownSubMenuProps,
+	DropdownItemProps,
+} from './types';
+
+// Observations / Questions:
+// - is it enough on the larger components to have only one forwarded ref?
+//   If we have only one, should it be to the "root", the "trigger", or the "content"?
+// - Should we be consistent in using the same value for the `asChild` prop on:
+//   - trigger
+//   - content
+//   - sub trigger
+//   - sub content
+// - Should we allow customizing the `asChild` prop on trigger and content?
+// - Props & customisability:
+//   - Which props should we expose?
+//   - Should props be "namespaced" for each subcomponent?
+//   - We should probably explicitly `Pick<>` every prop that we want to expose
+// - Subtrigger arrow:
+//   - Should it always be there (ie. an internal implementation)?
+//   - Should we just expect that the the consumers handle it themselves?
+//   - Should we expose it as a separate component that consumers could use?
+// - DropdowmMenuItem icon prop:
+//   - added to mirror previous menu item component
+//   - should we expect consumers to provide this directly with children?
+//   - should we expose prefix / suffix to help?
```

Comment:
- We should expose prefix / suffix props
- Standardize spacing and any other layout-related styles

Reply:
- Added `prefix`/`suffix` prop to `DropdownMenuItem`
- Added `suffix` prop to checkbox and radio items (no `prefix` for now, since these components may render an icon as their prefix already)